server/src/models/user.ts

```diff
@@ -1,8 +1,18 @@
+interface Admin {
+  isAdmin: boolean;
+  socket: string | null;
+}
+
 class User {
   public vote: number | null;
+  public admin: Admin;
 
-  constructor(public name: string, public isAdmin: boolean) {
+  constructor(public name: string, isAdmin: boolean) {
     this.vote = null;
+    this.admin = {
+      isAdmin: isAdmin,
+      socket: null,
+    };
```

Comment:
Proponuje nierobmy tutaj zagniezdzonego obiektu. Przechowjmy socket id dla kazdego usera i oprocz tego isAdmin: boolean.
Uprosci to kod, a socket id dla kazdego usera napewno sie przyda